### ISSUE TITLE
Dynamically set the prompt regex

### ISSUE THREAD

**Issue by rdnetto** [parsonsmatt/intero-neovim#79]
Many people users have a custom prompt set, but it's not obvious that the regex needs to be configured, or that that's the source of the problem (see #74).

Rather than relying on the assumption that the prompt contains a `>`, we should just parse `~/.ghci` and set it to something sane. Note that we'll need to handle the various escape sequences GHCi allows.

**Comment by parsonsmatt** [comment]
I'm not sure I'm a big fan of this. Users can set a prompt in `~/.ghci` or `./.ghci`, and they can also call `:set prompt "lmao"` at any point in the REPL. I'd rather set the REPL prompt on initialization to something we can control and discourage users from changing it.

```
Intero> your code here
```

seems like a reasonable thing IMO.

**Comment by rdnetto** [comment]
> I'm not sure I'm a big fan of this. Users can set a prompt in ~/.ghci or ./.ghci, and they can also call `:set prompt "lmao"` at any point in the REPL.

Iterating over the possible files should be fairly trivial. I agree that users manually setting the prompt will break this, but it seems like quite a corner case. (We could fix this easily by adding a handler for `:set prompt` similar to the existing one for `:reload` which populates quickfix, but I'm not sure if it's worth the effort).

> I'd rather set the REPL prompt on initialization to something we can control and discourage users from changing it.

It looks like the best way to do this would be to use [`--ghci-script`](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/ghci.html#the-ghci-files), which allows us to run a script after `.ghci`, etc.

The second approach is considerably simpler, but means users lose the ability to have a coloured prompt which matches their colour scheme. e.g. I use `:set prompt "\ESC[38;2;255;100;0mλ> \ESC[m"` to render `λ>` in orange. A better long-term solution for this use case would be to add a file type with syntax highlighting for the Intero session, which would also enable better highlighting of error messages. (Though maybe GHCi will colour these automatically in the future, making the point moot.)

tldr; forcing the prompt sounds good to me (though maybe to `λ>` for the screen space impaired), existing functionality/hacks broken by it have better solutions.
